libdnf/rpm/solv_sack.cpp

```diff
@@ -427,16 +426,13 @@ void SolvSack::Impl::make_provides_ready() {
     if (provides_ready)
         return;
     internalize_libsolv_repos();
-    Queue addedfileprovides;
-    Queue addedfileprovides_inst;
-    queue_init(&addedfileprovides);
-    queue_init(&addedfileprovides_inst);
-    pool_addfileprovides_queue(pool, &addedfileprovides, &addedfileprovides_inst);
-    if (addedfileprovides.count || addedfileprovides_inst.count) {
-        rewrite_repos(&addedfileprovides, &addedfileprovides_inst);
-    }
-    queue_free(&addedfileprovides);
-    queue_free(&addedfileprovides_inst);
+    solv::IdQueue addedfileprovides;
+    solv::IdQueue addedfileprovides_inst;
+    pool_addfileprovides_queue(pool, &addedfileprovides.get_queue(),
+                               &addedfileprovides_inst.get_queue());
+    if (addedfileprovides.size() || addedfileprovides_inst.size()) {
```

Comment:
`(addedfileprovides.size() == 0  || addedfileprovides_inst.size() == 0)` or `!empty()`. Oh, there is not `empty()` in `solv::IdQueue`. Add the method in future?